src/book/03-guides/01-main-codebase/02-maintaining-subsystems.mdx

```diff
@@ -0,0 +1,134 @@
+
+### NUpbr
+
+NUpbr is a Physically Based Rendering tool created in Blender. It creates semi-synthetic images with corresponding segmentation images for training.
+
+NUpbr requires Blender **2.79**. Download the appropriate file from the [Blender 2.79 Stable Release](https://download.blender.org/release/Blender2.79/).
+
+Clone the [NUpbr repository](https://github.com/NUbots/NUpbr) using
+
+```sh
+git clone https://github.com/NUbots/NUpbr.git
+```
+
+Navigate to the NUpbr directory and run
+
+```sh
+blender --python pbr.py
+```
+
+Find out more about NUpbr and how to edit the parameters on the [NUpbr NUbook page](/system/tools/nupbr).
+
+### NUgan
+
+The NUbots CycleGAN performs style transfer from the semi-synthetic style to a real image style. It creates more realistic images from the semi-synthetic images from NUpbr. It can be used for style transfer to a specific field for fine-tuning vision.
+
+Clone the [NUgan repository](https://github.com/NUbots/NUgan) using
+
+```sh
+git clone https://github.com/NUbots/NUgan.git
+```
+
+The [NUgan repository](https://github.com/NUbots/NUgan) details how to train, test and generate images.
+
+NUpbr images can be found on the NAS for the A datasets. The B dataset can be obtained by taking pictures, or collecting images from [Bit-Bots ImageTagger](https://imagetagger.bit-bots.de/).
+
+### Setting Up The Data
+
+The [Visual Mesh repository](https://github.com/Fastcode/VisualMesh) contains information on how to set up the datasets. It contains [a script](https://github.com/Fastcode/VisualMesh/blob/master/training/build_dataset.py) that takes data and converts it into the tfrecord format that the Visual Mesh expects. The script requires a folder of raw images, segmentation masks, and metadata. NUpbr will provide all these as output, and such data can be found on the NAS. If using output from NUgan, use the corresponding NUpbr segmentation masks and metadata, and add in the generated NUgan images in the image folder.
+
+## The Visual Mesh
+
+### Training
+
+Ensure you have followed the above steps to get a dataset to train on.
+
+As specified in the [Visual Mesh repository README](https://github.com/Fastcode/VisualMesh), install the dependencies from `requirements.txt`.
+
+```sh
+pip3 install -r requirements
+```
+
+Using [`cmake`](https://cmake.org/) and a C++ compiler with C++11 or later, build the custom TensorFlow op.
+
+```sh
+mkdir build
+cd build
+cmake ..
+make
+```
+
+A yaml file must be created with options for training. An example of this is in [the example_net.yaml file](https://github.com/Fastcode/VisualMesh/blob/master/example_net.yaml).
+
+Execute the training by running
+
+```sh
+./mesh.py train [config.yaml] [output_dir]
+```
+
+Where `[config.yaml]` is the options yaml file as specified above, and the output directory is the name of the directory you want the progress and results to be saved to. Progress can be monitored with TensorBoard.
```

Comment:
Put the config file in the output directory and run 
```
./mesh.py train [output_dir]
```

As I side note, the general convention that I am aware of for specifying command line parameters is (not sure if this is actually a thing, or just a thing that I like)
```
./mesh.py <required argument> [optional argument]
```

(also, run in docker)